### DISCUSSION TITLE
Timesketch SSO configuration

### DISCUSSION THREAD

**Discussion by qcshenderson** [google/timesketch#3098] [General]
Is there any documentation available for the configuration of SSO in Timesketch? There's nothing referencing it in the admin guide, etc.

**Comment by jkppr** [comment]
@qcshenderson the documentation on that topic is a bit sparse. But you can find some guidance on how to set-up SSO in the timesketch.conf comments: https://github.com/google/timesketch/blob/master/data/timesketch.conf#L52

Also the OSDFIR-Infrastructure project added some more information about usage of OIDC for login to their readme in their latests PR: https://github.com/google/osdfir-infrastructure/pull/144

**Reply by hendizzo** [reply]
Is it fair to say that SSO isn't a working feature of this at the moment @jkppr ?

**Reply by berggren** [reply]
Hi @hendizzo We do support SSO authenication via three different methods:

**1. Webserver based SSO (SAML for example):**
_Your web server can handle authentication for you by setting a environment
variable when the user is successfully authenticated. The standard environment
variable is REMOTE_USER and this is the default, but if your SSO system uses
another name you can configure that here._

**2. Google Cloud Identity-Aware Proxy (Cloud IAP) authentication configuration.**
_Cloud IAP controls access to your Timesketch server running on Google Cloud
Platform. Cloud IAP works by verifying a user’s identity and determining if
that user should be allowed to access the server._

**3. Google Cloud OpenID Connect (OIDC) authentication configuration.**
_Cloud OIDC controls access to your Timesketch server running on Google Cloud
Platform. Cloud OIDC works by verifying a user’s identity and determining if
that user should be allowed to access the server._
